crates/primitives/src/constants.rs

```diff
@@ -14,3 +19,26 @@ pub const SEC_NONCE_SIZE: usize = 64;
 
 /// The size (in bytes) of a Hash (such as [`Txid`](bitcoin::Txid)).
 pub const HASH_SIZE: usize = 32;
+
+/// A verifiably unspendable public key, produced by hashing a fixed string to a curve group
+/// generator.
+///
+/// Note that this is _not_ necessarily a uniformly-sampled curve point!
+///
+/// But this is fine; we only need a generator with no efficiently-computable discrete logarithm
+/// relation against the standard generator.
+pub static UNSPENDABLE_PUBLIC_KEY: LazyLock<XOnlyPublicKey> = LazyLock::new(|| {
+    XOnlyPublicKey::from_slice(sha256::Hash::hash(b"Strata unspendable").as_byte_array())
```

Comment:
Maybe make the `"Strata unspendable"` also a constant?

What if we want to use it somewhere in the future?

Reply:
I agree that this would be better, to make the use of magic values as obvious and clear as possible. Of course, in this case we probably want to _avoid_ reuse elsewhere, but same idea.